### PR TITLE
Add code to serialize plugin extensions

### PR DESCRIPTION
**Pull request by rudro**
- No new data model as the PluginizedComponent has all the info.

### CODE REVIEW

File: Generator/Sources/NeedleFramework/Generating/Pluginized/PluginExtensionSerializerTask.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    ///
+    /// - parameter component: The pluginized component that requires the
+    /// plugin extension provider.
+    init(component: PluginizedComponent) {
+        self.component = component
+    }
+
+    /// Execute the task and returns the data model.
+    ///
+    /// - returns: The `PluginExtensionProvider`.
```
Comment (neakor) [inline Generator/Sources/NeedleFramework/Generating/Pluginized/PluginExtensionSerializerTask.swift]: The `return` comment should be `SerializedProvider`.

File: Generator/Sources/NeedleFramework/Generating/Serializers/Pluginized/PluginExtensionClassNameSerializer.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// A serializer that produces the class name for the plugin extension.
```
Comment (neakor) [inline Generator/Sources/NeedleFramework/Generating/Serializers/Pluginized/PluginExtensionClassNameSerializer.swift]: This should be `A serializer that produces the class name for the plugin extension provider.`

File: Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift [hunk 3]
```diff
@@ -16,6 +16,14 @@
 
 import Foundation
 
+/// The data model representing a plugin extension provider to be generated for a
+/// specific pluginized component.
+struct PluginExtensionProvider {
+    let coreName : String
+    let nonCoreName : String
```
Comment (neakor) [inline Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift]: These should probably be named `pluginizedComponentName` and `nonCoreComponentName`.

File: Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift [hunk 4]
```diff
@@ -26,3 +34,15 @@ struct PluginExtension: Equatable {
     /// The list of properties.
     let properties: [Property]
 }
+
+
+/// A data model representing a plugin extension protocol of a NeedleFoundation
+/// `PluginizedComponent`.
```
Comment (neakor) [inline Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift]: The last bit should be `PluginExtension`.

File: Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift [hunk 5]
```diff
@@ -26,3 +34,15 @@ struct PluginExtension: Equatable {
...
+// This is separate from the `PluginizedComponent` data model, since its
+// plugin extension protocol may be declared in a separate file.
```
Comment (neakor) [inline Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift]: We can probably remove these two lines of comment?

File: Generator/Sources/NeedleFramework/Generating/Pluginized/PluginExtensionSerializerTask.swift (hunk 1 above)
Comment (rudro) [reply Generator/Sources/NeedleFramework/Generating/Pluginized/PluginExtensionSerializerTask.swift]: done

File: Generator/Sources/NeedleFramework/Generating/Serializers/Pluginized/PluginExtensionClassNameSerializer.swift (hunk 2 above)
Comment (rudro) [reply Generator/Sources/NeedleFramework/Generating/Serializers/Pluginized/PluginExtensionClassNameSerializer.swift]: done

File: Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift (hunk 3 above)
Comment (rudro) [reply Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift]: unused struct, deleted.

File: Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift (hunk 4 above)
Comment (rudro) [reply Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift]: done

File: Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift (hunk 5 above)
Comment (rudro) [reply Generator/Sources/NeedleFramework/Models/Pluginized/PluginExtension.swift]: done

**Pull request merged**